bindings/java/src/org/sleuthkit/datamodel/OsAccountRealmManager.java

```diff
@@ -775,11 +861,9 @@ void mergeRealms(OsAccountRealm sourceRealm, OsAccountRealm destRealm, CaseDbTra
 		
 		// Update the destination realm if it doesn't have the name or addr set and the source realm does
 		if (!destRealm.getRealmAddr().isPresent() && sourceRealm.getRealmAddr().isPresent()) {
-			destRealm.setRealmAddr(sourceRealm.getRealmAddr().get());
-			updateRealm(destRealm, trans.getConnection());
+			updateRealm(destRealm, sourceRealm.getRealmAddr().get(), null, trans.getConnection());
 		} else if (destRealm.getRealmNames().isEmpty() && !sourceRealm.getRealmNames().isEmpty()) {
-			destRealm.addRealmName(sourceRealm.getRealmNames().get(0));
-			updateRealm(destRealm, trans.getConnection());
+			updateRealm(destRealm, null, sourceRealm.getRealmNames().get(0), trans.getConnection());
```

Comment:
Do these lines mean that we are updating the signature twice? If so, can we combine operations by using more conditional clauses?

Reply:
We're actually only updating the signature up above this (makeMergedRealmSignature()). In updateRealm() it will stay the same because the status isn't ACTIVE. The general idea is that we enforce in the SQL update statement that the signature won't change once the status is MERGED instead of going by anything in the realm object to try to avoid timing issues.